packages/angular/src/2016-12-01/button-directive.js

```diff
@@ -0,0 +1,58 @@
+/**
+Copyright 2016 Autodesk,Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+const { ColorValidator, SizeValidator } = require('./validators');
+require('@orion-ui/components/lib/2016-12-01/button');
+
+// Use ng-disabled and ng-click
+const ButtonDirective = ($log) => {
+  const colorValidator = new ColorValidator();
+  const sizeValidator = new SizeValidator();
+  const configurable = ['color', 'background', 'size'];
+
+  const validations = {
+    color: colorValidator,
+    background: colorValidator,
+    size: sizeValidator,
+  };
```

Comment:
Why are these validations happening at the Angular level? Wouldn't these be better served at the Web Component level since they aren't angular specific?